packages/studio-ui/src/components/CollapsibleSidebar.tsx

```diff
@@ -0,0 +1,40 @@
+import classNames from "classnames";
+import { PropsWithChildren, useCallback, useState } from "react";
+import { ReactComponent as Hamburger } from "../icons/hamburger.svg";
+
+interface CollapsibleSidebarProps {
+  side: "right" | "left";
+}
+export default function CollapsibleSidebar({
+  side,
+  children,
+}: PropsWithChildren<CollapsibleSidebarProps>): JSX.Element {
+  const [open, setOpen] = useState(true);
+  const toggleOpen = useCallback(() => {
+    setOpen(!open);
+  }, [open]);
+
+  const justifyButtonStyle = classNames("flex", {
+    "justify-start": side === "right",
+    "justify-end": side === "left",
+  });
+
+  const sidebarStyle = classNames("flex flex-col grow", {
+    hidden: !open,
+  });
+
+  return (
+    <div
+      className={classNames("flex flex-col", {
```

Comment:
nit: can we do the `classNames` logic in the component body like how we compute the other styles?

Reply:
yes, done!